src/tapedfunction.jl

```diff
@@ -101,26 +144,9 @@ function (instr::Instruction{typeof(_new)})()
     end
 end
 
+## internal functions
 
-function increase_counter!(t::Tape)
-    t.counter > length(t) && return
-    # instr = t[t.counter]
-    t.counter += 1
-    return t
-end
-
-function run(tape::Tape, args...)
-    if length(args) > 0
-        input = map(box, args)
-        tape[1].input = input
-    end
-    for instruction in tape
-        instruction()
-        increase_counter!(tape)
-    end
-end
-
-function run_and_record!(tape::Tape, f, args...)
+function run_and_record!(tape::Taped, f, args...)
```

Comment:
I think `run_and_record!` is a bit confusing actually, because it could mean we are running the entire `tape`. Maybe rename this funciuton to 
```suggestion
function track!(tape::Taped, f, args...)
```

Reply:
done.